src/main/java/ai/ancf/lmos/operator/service/AgentServiceQuery.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package ai.ancf.lmos.operator.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@org.springframework.stereotype.Service
+public class AgentServiceQuery {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AgentServiceQuery.class);
+
+    private final WebClient webClient;
+
+    public AgentServiceQuery() {
+        this.webClient = WebClient.builder().build();
+    }
+
+    public String queryAgentService(String serviceUrl) {
+        return webClient.get()
+            .uri(serviceUrl + "/agent")
+            .accept(MediaType.APPLICATION_JSON)
+            .retrieve()
+            .bodyToMono(String.class).block();
```

Comment:
We could parse this into a Thing Description to validate it here, or?

Reply:
resolved